shared/rakefiles/xk.rake

```diff
@@ -76,17 +76,10 @@ task :refresh_common_infra, [:project_type] => [@gcp_creds_file] do | taskname,
   # conflicts at the creation resources we need to import the already created
   # DNS zone.
 
-  output = `#{@exekube_cmd} sh -c "\
-    terragrunt output -json dns_zones \
+  dns_zone_found = `#{@exekube_cmd} sh -c "\
+    terragrunt state show module.gke_network.google_dns_managed_zone.dns_zones \
     --terragrunt-working-dir /project/live/#{@env}/infra/network 2>/dev/null \
     "`
-  begin
-    # What does !! mean?
-    # https://stackoverflow.com/questions/524658/what-does-mean-in-ruby
-    dns_zone_found = !!JSON.parse(output)
-  rescue JSON::ParserError
-    dns_zone_found = false
-  end
 
   unless dns_zone_found
```

Comment:
I don't think this will ever trigger. I assume terragrunt behaves in a same way as Terrafrom, and in case of a missing resource in TF state, TF returns empty `stdout`. In Ruby result of call made using backticks is `stdout` of given command.

Seems like string, even empty, always evaluates to `true`:
```
irb(main):053:0> !!""
=> true
irb(main):054:0> !!"abc"
=> true
````
-> result of `unless dns_zone_found` will be that the first block will be never applied.

Also I find using `unless` together with `else` very confusing. Why not reverse the logic to `if` and apply the second block first, followed by `else` applying the current `unless` block?

Reply:
> Also I find using unless together with else very confusing. Why not reverse the logic to if and apply the second block first, followed by else applying the current unless block?

+1 on this comment. I find `unless` to be most useful when used in a postfix way for single-liners. When used in conjunction with `else`, it can get confusing.